### PR TITLE
Add external link

### PR DESCRIPTION
**Pull request by chhaymenghong**

### CODE REVIEW

File: src/directives/mmsViewLink.js [hunk 1]
```diff
@@ -148,7 +148,8 @@ function mmsViewLink(ElementService, UtilsService, $compile, growl, ViewService,
             suppressNumbering: '<'
         },
         require: ['?^^mmsCf', '?^^mmsView'],
-        template: '<a ng-if="!loading" target="{{target}}" ng-class="linkClass" ng-href="{{href}}"><i ng-class="linkIconClass" aria-hidden="true"></i><span ng-if="linkText">{{linkText}}</span><span ng-if="!linkText && change">{{type}}{{element._veNumber}}{{suffix}}</span><span ng-if="!linkText && !change">{{name || "Unnamed View"}}</span></a>',
+        template: '<span ng-if="!loading"><a target="{{target}}" ng-class="linkClass" ng-href="{{href}}"><i ng-class="linkIconClass" aria-hidden="true"></i><span ng-if="linkText">{{linkText}}</span><span ng-if="!linkText && change">{{type}}{{element._veNumber}}{{suffix}}</span><span ng-if="!linkText && !change">{{name || "Unnamed View"}}</span></a>' +
+        '<a class="external-link" target=_blank ng-href="{{href}}" ng-if="linkText"><i class="fa fa-external-link" aria-hidden="true"></i></a></span>',
```
Comment (dlamoris) [inline src/directives/mmsViewLink.js]: add a no-print class to the external link too

**Comment by dlamoris** [comment]
are we sure we want to do this? this will add the external link button to all view links in a doc too, when they can already right click on them.

**Comment by ellenvanwyk** [comment]
It should just be for the open doc link. Make it a bit easier for people who don't know to right click.

**Comment by dlamoris** [comment]
in that case, mmsViewLink shouldn't be changed, just the mmsView template where mmsLink is true, to add the open in new tab (only the doc preview page)

**Comment by chhaymenghong** [comment]
hmm. It wont "add the external link button to all view links in a doc". It will only add that button to the doc preview page because of the "linkText" conditional which is only set in our entire app for that case. That was one of the reasons I decided not to add the button under mmsview template. Another reason is that, if i were to add it under mmsview template, i have to also add extra logic to generate href value whereas in another case i can just reuse it. As for adding "no-print", that sounds good. Actually I am going to add that to the "Open Document" button as well because it doesn't have any at the moment.

**Comment by dlamoris** [comment]
The user can still populate linktext manually any place there’s a view link, it’s better to instead add another parameter for adding the external link so it’s explicit

**Comment by chhaymenghong** [comment]
cool. I will add a separate parameter.

**Pull request merged**